dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -596,4 +652,73 @@ public static void saveMultiTree(MultiTree o, long languageId) {
 			throw new DotRuntimeException(e.toString());
 		}
 	}	
+	
+    /**
+     * Update a HTML Page Version Info Timestamp given a HTMLPage Identifier.
+     * If a MultiTree Object has been added or deleted from this page,
+     * its version_ts value needs to be updated so it can be included
+     * in future Push Publishing tasks
+     * 
+     * @param id The HTMLPage Identifier to pass in 
+     * @throws DotContentletStateException
+     * @throws DotDataException 
+     * @throws DotSecurityException 
+     *            
+     */
+    private static void updateVersionTs(String id) throws DotStateException, DotDataException, DotSecurityException {
+        updateVersionTs(id, APILocator.getLanguageAPI().getDefaultLanguage().getId());    
+    }
+	
+    /**
+     * Update a HTML Page Version Info Timestamp given a HTMLPage Identifier and a Language Id.
+     * If a MultiTree Object has been added or deleted from this page,
+     * its version_ts value needs to be updated so it can be included
+     * in future Push Publishing tasks
+     * 
+     * @param id The HTMLPage Identifier to pass in 
+     * @throws DotContentletStateException
+     * @throws DotDataException 
+     * @throws DotSecurityException 
+     *            
+     */
+	private static void updateVersionTs(String id, Long languageId) throws DotStateException, DotDataException, DotSecurityException {
+	    Identifier ident = APILocator.getIdentifierAPI().find(id);
+        ContentletVersionInfo versionInfo = APILocator.getVersionableAPI()
+                .getContentletVersionInfo(ident.getId(), languageId);
+        versionInfo.setVersionTs(new Date());
+        APILocator.getVersionableAPI().saveContentletVersionInfo(
+                versionInfo);
+	}
+	
+    /**
+     * Refresh Cached objects of page given a HTMLPage Identifier.
+     * 
+     * @param id The HTMLPage Identifier to pass in 
+     * @throws DotContentletStateException
+     * @throws DotDataException 
+     * @throws DotSecurityException 
+     *            
+     */
+    private static void refreshPageInCache(String id) throws DotContentletStateException, DotDataException, DotSecurityException {
+        refreshPageInCache(id, APILocator.getLanguageAPI().getDefaultLanguage().getId());
+    }
+    
+
+    /**
+     * Refresh Cached objects of page given a HTMLPage Identifier and a Language Id.
+     * 
+     * @param id The HTMLPage Identifier to pass in 
+     * @param languageId A language Id that points to a specific version we'll clean up
+     * @throws DotContentletStateException
+     * @throws DotDataException 
+     * @throws DotSecurityException 
+     *            
+     */
+    private static void refreshPageInCache(String id, Long languageId) throws DotContentletStateException, DotDataException, DotSecurityException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.portlets.contentlet.business.DotContentletStateException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)